version: "4"

services:
  postgres:
    image: postgres:16.0
    container_name: mayhem_database
    environment:
      POSTGRES_DB: "mayhem"
      POSTGRES_USER: "biba"
      POSTGRES_PASSWORD: "boba"
      PGDATA: "/postgresql/data/pgdata"
    volumes:
      - ../2. Init Database:/docker-entrypoint-initdb.d
      - mayhem-data:/postgresql/data/
#      - .:/docker-entrypoint-initdb.d
    ports:
      - "5432:5432"
    deploy:
      resources:
        limits:
          cpus: '1'
          memory: 4G
    networks:
      - postgres
  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4:7.2
    environment:
      PGADMIN_DEFAULT_EMAIL: "aboba@mail.ru"
      PGADMIN_DEFAULT_PASSWORD: "biba"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - pgadmin-data:/pgadmin
    ports:
      - "5050:80"
    restart: unless-stopped
    deploy:
      resources:
        limits:
          cpus: '0.5'
          memory: 1G
    networks:
      - postgres
  tg_bot:
    container_name: tg_bot
    build:
      context: ./
      dockerfile: Dockerfile
    volumes:
      - ./:/app
    expose:
      - 3000
    ports:
      - "3000:3000"
    command: >
      /bin/sh -c 'python main.py'

volumes:
  mayhem-data:
  pgadmin-data:

networks:
  postgres:
    driver: bridge
